FrameworkSystem/Client/ProxyManagerClient.py

```diff
@@ -147,49 +184,47 @@ def uploadProxy(self, proxy=False, diracGroup=False, chainToConnect=False, restr
       if not result['OK']:
         return S_ERROR("Can't load %s: %s " % (proxyLocation, result['Message']))
 
-    if not chainToConnect:
-      chainToConnect = chain
-
     # Make sure it's valid
-    if chain.hasExpired()['Value']:
+    if chain.hasExpired().get('Value'):
```

Comment:
Did you do this change because you encounter an issue there ? I know it wasn't the case before, but in practice we do not test the case of `S_ERROR` being returned

Reply:
How I remember, this change is due to an KeyError: 'Value'

Reply:
Then you should also test the flag `['OK']` and return in case of error

Reply:
fixed